File: Mcms.Api.Business.Poco/Models/Comments/Comment.cs
using Mcms.Api.Business.Poco.Models.Core.Release;
using Mcms.Api.Business.Poco.Models.Mapping.Mappings;

namespace Mcms.Api.Business.Poco.Models.Comments
{

    /// <summary>
    /// Represents a single comment made with in the system.
    /// </summary>
    public class Comment
    {
        /// <summary>
        /// The id of the comment.
        /// </summary>
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        /// <summary>
        /// The id of the user who created the comment.
        /// </summary>
        [Required]
        public Guid CreatedBy { get; set; }

        /// <summary>
        /// The moment the comment was created.
        /// </summary>
        [Required]
        public DateTime CreatedOn { get; set; }

        /// <summary>
        /// The content of the comment.
        /// </summary>
        [Required]
        public string Content { get; set; }

        /// <summary>
        /// The reactions applied to the comment.
        /// </summary>
        public List<CommentReaction> Reactions { get; set; }

        /// <summary>
        /// Indicates if the comment has been edited.
        /// </summary>
        public bool HasBeenEdited { get; set; }

        /// <summary>
        /// Indicates if the comment has been deleted.
        /// </summary>
        public bool IsDeleted { get; set; }

        /// <summary>
        /// The user who deleted the comment, can be the creating user or any user with enough rights.
        /// </summary>
        public Guid DeletedBy { get; set; }

        /// <summary>
        /// The proposed mapping this comment was made on.
        /// </summary>
        public ProposedMapping ProposedMapping { get; set; }

        /// <summary>
        /// The release this comment was made on.
        /// </summary>
        public Release Release { get; set; }
    }
}

File: Development/API/WebApi/src/main/java/com/mcms/api/webapi/data/models/comments/Comment.java
package com.mcms.api.webapi.data.models.comments;

import java.sql.Timestamp;
import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.PersistenceConstructor;
import org.springframework.data.relational.core.mapping.Table;

@Table
public class Comment {

    @Id
    private final UUID id;
    private final UUID createdBy;
    private final Timestamp createdOn;
    private final String content;
    private final boolean hasBeenEdited;
    private final boolean isDeleted;
    private final UUID deletedBy;
    private final Timestamp deletedOn;
    private final UUID proposedMappingId;
    private final UUID releaseId;
    private final UUID parentCommentId;

    @PersistenceConstructor
    Comment(
                    final UUID id,
                    final UUID createdBy,
                    final Timestamp createdOn,
                    final String content,
                    final boolean hasBeenEdited,
                    final boolean isDeleted,
                    final UUID deletedBy,
                    final Timestamp deletedOn,
                    final UUID proposedMappingId,
                    final UUID releaseId,
                    final UUID parentCommentId) {
        this.id = id;
        this.createdBy = createdBy;
        this.createdOn = createdOn;
        this.content = content;
        this.hasBeenEdited = hasBeenEdited;
        this.isDeleted = isDeleted;
        this.deletedBy = deletedBy;
        this.deletedOn = deletedOn;
        this.proposedMappingId = proposedMappingId;
        this.releaseId = releaseId;
        this.parentCommentId = parentCommentId;
    }


}
